dev/TEAL_opcodes.md

```diff
@@ -730,6 +764,46 @@ When A is a uint64, index 0 is the least significant bit. Setting bit 3 to 1 on
 - pop a byte-array A, integer B, and small integer C (between 0..255). Set the Bth byte of A to C, and push the result
 - LogicSigVersion >= 3
 
+## extract s l
+
+- Opcode: 0x57 {uint8 start position} {uint8 length}
+- Pops: *... stack*, []byte
+- Pushes: []byte
+- pop a byte-array A. For immediate values in 0..255 S and L: extract a range of bytes from A starting at S up to but not including S+L, push the substring result. If L is 0, then extract to the end of the string. If S or S+L is larger than the array length, the program fails
+- LogicSigVersion >= 5
+
+## extract3
+
+- Opcode: 0x58
+- Pops: *... stack*, {[]byte A}, {uint64 B}, {uint64 C}
+- Pushes: []byte
+- pop a byte-array A and two integers B and C. Extract a range of bytes from A starting at B up to but not including B+C, push the substring result. If B+C is larger than the array length, the program fails
+- LogicSigVersion >= 5
+
+## extract16bits
```

Comment:
Maybe I would call that `extractUint16` rather than `extract16bits`
as really we're extracting an integer.
Same comment for the ones below.

Reply:
I'd prefer not to.  We don't normally encode the return type in the name.  I admit there's slightly more room for confusion here, but the camelCase is kind of ugly.  I guess I could see extract_uint16 if you feel strongly.  I think I was worried that people might think `extract_unit16 3` meant the 3rd uint16, as opposed to the uint16 starting at byte 3.  But I don't know that I think that now.

Reply:
I got confused with the caml case.
I prefer `extract_uint16` but if you feel strongly against it, we can keep `extract16bits`.

Reply:
done